lib/llvm-backend/src/code.rs

```diff
@@ -4472,10 +4564,36 @@ impl<'ctx> FunctionCodeGenerator<CodegenError> for LLVMFunctionCodeGenerator<'ct
                     builder.build_float_to_signed_int(v1, intrinsics.i32_ty, &state.var_name());
                 state.push1(res);
             }
-            Operator::I32TruncSSatF32 | Operator::I32TruncSSatF64 => {
-                let v1 = state.pop1()?.into_float_value();
-                let res =
-                    builder.build_float_to_signed_int(v1, intrinsics.i32_ty, &state.var_name());
+            Operator::I32TruncSSatF32 => {
+                let (v, i) = state.pop1_extra()?;
+                let v = apply_pending_canonicalization(builder, intrinsics, v, i);
+                let v = v.into_float_value();
+                let res = trunc_sat_scalar(
+                    builder,
+                    intrinsics.i32_ty,
+                    std::i32::MIN as u64,
+                    2147483520u64, // bits as f32: 0x4effffff
+                    std::i32::MIN as u32 as u64,
+                    std::i32::MAX as u32 as u64,
```

Comment:
⛳️ `i32::max_value()` doesn't need `std::` 🤔 either way is fine though

Reply:
Yeah, I'm fine with either -- the only reason I mention it is that I haven't seen the MIN and MAX constants in a while.  Now that I see them again I do remember having seen them in the past.